### PR TITLE
Addlognormal

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @kortschak @jonlawlor

File: dist/distribution_test.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	"sort"
+	"testing"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/stat"
+)
+
+// fullDist is a distribution that implements all the basic functions.
+type fullDist interface {
+	CDF(x float64) float64      // check
```
Comment (kortschak) [inline dist/distribution_test.go]: Are these comments notes to you?

File: dist/distribution_test.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	Skewness() float64          // check
+	StdDev() float64            // check
+	Survival(x float64) float64
+	Variance() float64 // check
+}
+
+// testFullDist tests all of the functions of a fullDist.
+func testFullDist(t *testing.T, f fullDist, i int) {
+	tol := 1e-1
+	n := 1000000
```
Comment (kortschak) [inline dist/distribution_test.go]: const n = 1e6

**Comment by kortschak** [comment]
LGTM

File: dist/distribution_test.go (hunk 2 above)
Comment (btracey) [inline dist/distribution_test.go]: Ah. Tricksy. Constants are by far the most confusing thing in Go.

File: dist/distribution_test.go (hunk 1 above)
Comment (btracey) [inline dist/distribution_test.go]: Sorry, yea, just keeping track of what I had tested.

File: dist/distribution_test.go (hunk 2 above)
Comment (kortschak) [inline dist/distribution_test.go]: Yeah, the reasoning is that it's easier to read 1e6 than 1000000, but if you do 1e6 it's declared as float64, so you need to make it untyped or `var n int = 1e6` or `n := int(1e6)`, neither of which look that good.

Comment (btracey) [inline dist/distribution_test.go]: > On Jun 2, 2015, at 5:15 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In dist/distribution_test.go <https://github.com/gonum/stat/pull/42#discussion_r31584347>:
> 
> > +	LogProb(x float64) float64  // check
> > +	Mean() float64              // check
> > +	Median() float64            // check
> > +	Prob(x float64) float64     // check
> > +	Quantile(p float64) float64 // check
> > +	Rand() float64              // check
> > +	Skewness() float64          // check
> > +	StdDev() float64            // check
> > +	Survival(x float64) float64
> > +	Variance() float64 // check
> > +}
> > +
> > +// testFullDist tests all of the functions of a fullDist.
> > +func testFullDist(t *testing.T, f fullDist, i int) {
> > +	tol := 1e-1
> > +	n := 1000000
> Yeah, the reasoning is that it's easier to read 1e6 than 1000000, but if you do 1e6 it's declared as float64, so you need to make it untyped or var n int = 1e6 or n := int(1e6), neither of which look that good.
> 
Yea, I worked through all of those logical steps, but hadn’t thought of the simpler “const” option.

**Comment by jonlawlor** [comment]
Late LGTM.

**Pull request merged**